modules/index/src/test/java/org/apache/ignite/internal/index/IndexBuildingStarterTaskTest.java

```diff
@@ -0,0 +1,349 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/** For {@link IndexBuildingStarterTask} testing. */
+@ExtendWith(MockitoExtension.class)
+public class IndexBuildingStarterTaskTest extends IgniteAbstractTest {
+    private static final IndexMessagesFactory FACTORY = new IndexMessagesFactory();
+
+    private final IgniteSpinBusyLock busyLock = new IgniteSpinBusyLock();
+
+    private final HybridClock clock = new HybridClockImpl();
+
+    private final ClockWaiter clockWaiter = spy(new ClockWaiter(NODE_NAME, clock));
+
+    private final CatalogManager catalogManager = createTestCatalogManager(NODE_NAME, clockWaiter);
+
+    private final ExecutorService executor = spy(newSingleThreadExecutor());
+
+    private final ClusterService clusterService = createClusterService();
+
+    @Mock
+    private PlacementDriver placementDriver;
+
+    @Mock(strictness = LENIENT)
+    private LogicalTopologyService logicalTopologyService;
+
+    private IndexBuildingStarterTask task;
+
+    private CatalogIndexDescriptor indexDescriptor;
+
+    @BeforeEach
+    void setUp() {
+        assertThat(allOf(clockWaiter.start(), catalogManager.start()), willCompleteSuccessfully());
+
+        createTable(catalogManager, TABLE_NAME, COLUMN_NAME);
+        createIndex(catalogManager, TABLE_NAME, INDEX_NAME, COLUMN_NAME);
+
+        indexDescriptor = indexDescriptor(catalogManager, INDEX_NAME, clock);
+
+        CompletableFuture<ReplicaMeta> localNodeReplicaMetaFuture = completedFuture(
+                createLocalNodeReplicaMeta(HybridTimestamp.MIN_VALUE, HybridTimestamp.MAX_VALUE)
+        );
+
+        when(placementDriver.awaitPrimaryReplica(any(), any(), anyLong(), any())).thenReturn(localNodeReplicaMetaFuture);
+
+        CompletableFuture<LogicalTopologySnapshot> logicalTopologySnapshotFuture = completedFuture(
+                new LogicalTopologySnapshot(1, List.of(LOGICAL_LOCAL_NODE))
+        );
+
+        when(logicalTopologyService.logicalTopologyOnLeader()).thenReturn(logicalTopologySnapshotFuture);
+
+        task = new IndexBuildingStarterTask(
+                indexDescriptor.id(),
+                indexDescriptor.tableId(),
+                catalogManager,
+                placementDriver,
+                clusterService,
+                logicalTopologyService,
+                clock,
+                clockWaiter,
+                executor,
+                busyLock
+        );
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        closeAll(
+                catalogManager::beforeNodeStop,
+                clockWaiter::beforeNodeStop,
+                catalogManager::stop,
+                clockWaiter::stop,
+                task == null ? null : task::stop
+        );
+
+        shutdownAndAwaitTermination(executor, 1, SECONDS);
+    }
+
+    @Test
+    void testSimpleTaskExecution() {
+        clearInvocations(clockWaiter);
```

Comment:
This should be done automatically by `MockitoExtension`, so this seems redundant. Or maybe you need to annotate the field with `@Spy` (and remove explicit `spy(`)

Reply:
fix it